pynicotine/users.py

```diff
@@ -116,20 +118,86 @@ def request_set_status(self, status):
     def request_user_stats(self, username):
         core.send_message_to_server(slskmessages.GetUserStats(username))
 
-    def watch_user(self, username):
-        """Tell the server we want to be notified of status/stat updates for a
-        user."""
+    def watch_user(self, username, context=None, is_implicit=False):
+        """Tells the server we want to be notified of status updates for a
+        user.
+
+        context is a string specifying where the user is being watched.
+        The same context must be provided when calling unwatch_user().
+
+        is_implicit is set when receiving status updates from the server
+        without sending a message first. At present, this only happens for
+        users in a joined chat room.
+        """
 
         if self.login_status == slskmessages.UserStatus.OFFLINE:
             return
 
-        if username in self.watched:
+        watched_user = self.watched.get(username)
+
+        if watched_user is None:
+            self.watched[username] = watched_user = WatchedUser(username)
+
+        if not context:
+            context = "unknown"
```

Comment:
Why is this needed?

If the function call is made improperly, then an error message should be logged.

Reply:
I'd like to get the change into 3.3.x, but wasn't sure if any plugins are calling this function. I'll just mark watch_user() with no context as deprecated, and make the context mandatory in 3.4.0.